src/commands/run.rs

```diff
@@ -360,6 +360,25 @@ impl RunCommand {
 
         Ok(())
     }
+
+    fn load_module(&self, engine: &Engine, path: &Path) -> Result<Module> {
+        let contents = std::fs::read(path)
+            .with_context(|| format!("failed to read path: {}", path.display()))?;
+
+        // Part of the CLI is that we can load precompiled wasm modules output
+        // by `wasmtime compile`. This is the raw output of `module.serialize()`
+        // which for now is an ELF image, so we detect whether this is an ELF
+        // file and if so try to deserialize a module from that.
+        //
+        // This technically makes this whole implementation unsafe because it's
+        // not safe to feed arbitrary bytes into `deserialize`, but that's part
+        // of the CLI interface, so it's just how this is designed.
```

Comment:
The versioning check does give us a small amount of additional safety, since that has to pass before we bincode::deserialize. Maybe worth mentioning?

Reply:
Yeah the main thing we're worried about with `Module::deserialize` is that it's not safe to pass user-defined input into it because it's easy to construct an ELF file which executes arbitrary things but passes our deserialization. I do think, though, that we have enough checks in place to catch accidental usage without segfaulting.